pyleecan/Functions/Save/save_json.py

```diff
@@ -230,7 +251,12 @@ def split_obj_dict(cls_tupel, obj_dict, folder, split_list, file_ext, logger):
                 cls_tupel, val, folder, split_list, file_ext, logger
             )
 
-        if "__class__" in obj_dict.keys() and obj_dict["__class__"] in cls_tupel:
+        if (
+            "__class__" in obj_dict.keys()
+            and obj_dict["__class__"] in cls_tupel
+            and "name" in obj_dict.keys()
+            and obj_dict["name"] not in ["", None]
```

Comment:
Hello @BonneelP ,

with these lines you prevent to split unnamed and nameless objects. So it's impossible to e.g. split Input, EEC or Drive objects. 
I guess you told why you did that, but I missed it :-) Maybe there should be another input arg 'save_named_only=False' if you need that restriction.

Best regards, Sebastian

Reply:
Hello,

The purpose was to avoid splitting material with name at "None" (which most of the time means that the material is not defined). I will change the condition to always split the the object without name property. By the way, I will introduce a "name" property in SlotUD and HoleUD to split them as well.
Regarding the json indentation, it is worth the try. I will check what an indentation of 2 looks like.

Best regards,
Pierre